packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+  useRef,
+} from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import useApi from '../../../../api/useApi';
+
+/**
+ * Hook used for taxonomy filters logic.
+ * Initializes the taxonomy filters data.
+ *
+ * @return {Array} [taxonomies, queryTaxonomyTerms] initial taxonomy filters data and a function to query the terms.
+ */
+
+function useTaxonomyFilters() {
+  const [taxonomies, setTaxonomies] = useState([]);
+  const termPromises = useRef([]);
+
+  const { getTaxonomies, getTaxonomyTerms } = useApi(
+    ({
+      actions: {
+        taxonomyApi: { getTaxonomies, getTaxonomyTerms },
+      },
+    }) => ({
+      getTaxonomies,
+      getTaxonomyTerms,
+    })
+  );
+
+  /**
+   * Maps the fetched taxonomy data back to parent taxonomy.
+   *
+   * @return {void}
+   */
+  const _setTaxonomiesTermData = useCallback(async () => {
+    const fetched = await Promise.all(termPromises.current);
+    const taxonomyObject = {};
+    for (const arr of fetched) {
+      // grab the first elements 'taxonomy' which should map to the parents 'slug'
+      const key = arr.at(0)?.taxonomy;
+      if (key) {
+        taxonomyObject[key] = arr;
+      }
+    }
+    setTaxonomies((current) => {
+      return current.map((c) => {
+        const data = taxonomyObject[c.slug] || [];
+        return {
+          ...c,
+          data,
+        };
+      });
+    });
```

Comment:
```suggestion
    setTaxonomies((current) => current.map((c) => {
          ...c,
          data: taxonomyObject[c.slug] || [],
        }
    });
```

Allow function do no need returns right? 

Also, current, isnt a great name, how about `taxonomies`?

Reply:
`taxonomies` is already declared in the upper scope, I can rename this to be more readable.